packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2547,13 +2543,16 @@ function throwAndUnwindWorkLoop(
   }
 }
 
-function panicOnRootError(error: mixed) {
+function panicOnRootError(root: FiberRoot, error: mixed) {
   // There's no ancestor that can handle this exception. This should never
   // happen because the root is supposed to capture all errors that weren't
   // caught by an error boundary. This is a fatal error, or panic condition,
   // because we've run out of ways to recover.
   workInProgressRootExitStatus = RootFatalErrored;
-  workInProgressRootFatalError = error;
+  logCapturedError(
+    root.current,
+    createCapturedValueAtFiber(error, root.current),
+  );
```

Comment:
Instead of stashing this and then logging, I just log it directly. I think that's fine since nothing else is really supposed to be happening in between but not sure.